CRM/Contact/Form/ProfileContact.php

```diff
@@ -75,37 +58,81 @@ public static function preProcess(&$form) {
    * @param CRM_Core_Form $form
    *
    */
-  public static function buildQuickForm(&$form) {
-    $ufGroup = new CRM_Core_DAO_UFGroup();
-    $ufGroup->id = $form->_honoreeProfileId;
-    if (!$ufGroup->find(TRUE)) {
-      CRM_Core_Error::fatal(ts('Chosen honoree profile for this contribution is disabled'));
+  public function buildQuickForm() {
+    $contactID = $this->getContactID();
+
+    if ($this->_prefix == 'onbehalf') {
+      $this->assign('fieldSetTitle', ts('Organization Details'));
+      $this->assign('profileId', $this->_profileId);
+
+      if ($contactID) {
+        $employer = CRM_Contact_BAO_Relationship::getPermissionedEmployer($contactID);
```

Comment:
I seem to have lost the rest of the patch - but I think it makes sense to rename this function & pass relationship_type as a parameter as part of breaking the hard-codedness 

https://github.com/eileenmcnaughton/civicrm-core/commit/286bf5c08b75a57deef616f721d7fe3b9189e518